src/Compilers/VisualBasic/Portable/Semantics/OverloadResolution.vb

```diff
@@ -4080,15 +4400,17 @@ Bailout:
                     '       This rule also applies to the types that extension methods are defined on.
                     '7.2.	If M and N are extension methods and the target type of M is a class or
                     '       structure and the target type of N is an interface, eliminate N from the set.
-                    If ShadowBasedOnReceiverType(existingCandidate, newCandidate, existingWins, newWins, useSiteInfo) Then
+                    If (Not someCandidatesHaveOverloadResolutionPriority OrElse
+                        (signatureMatch AndAlso Not (existingCandidate.Candidate.IsExtensionMethod OrElse newCandidate.Candidate.IsExtensionMethod))) AndAlso
```

Comment:
What effect does this `OrElse` clause have, and are we testing it?

Reply:
> What effect does this ```OrElse``` clause have, and are we testing it?

EarlyFilteringOnReceiverType_01 fails if we remove it. The meaning of it is pretty much similar to what the comment before the previous `If` says. I.e. priority filtering for extension methods must come before filtering by receiver. Because for extension methods the filtering by receiver is not about shadowing by signature between methods declared in base and derived types.

Reply:
It looks like `EarlyFilteringOnReceiverType_01` still passes of the entire `OrElse ...` clause is removed.

Reply:
I am not sure what you did, but when I remove ``` AndAlso Not (existingCandidate.Candidate.IsExtensionMethod OrElse newCandidate.Candidate.IsExtensionMethod)``` part. The test fails for me. Perhaps that should be taken offline.

Reply:
As we discussed, I was referring to the entire `OrElse (signatureMatch AndAlso Not (...))`.